database/database.go

```diff
@@ -1,127 +1,585 @@
+		unsafeAddTeacherToLocalDatabase(t)
+	}
+
+	rows.Close()
+
+	log.Print("Done")
+	return nil
+}
+
+// Just adds quote to localDatabase (quoteSlice and wordsMap) without checking q.QuoteID
+// using addQuoteToLocalDatabase without checking if q.QuoteID already exists may be fatal
+func addQuoteToLocalDatabase(q QuoteT) error {
+
+	localDatabase.mux.LockWrite()
+	defer localDatabase.mux.UnlockWrite()
+
+	localDatabase.quoteSlice = append(localDatabase.quoteSlice, q)
+	var enumid int32 = int32(len(localDatabase.quoteSlice) - 1)
+
+	if enumid < 0 {
+		return errors.New("At addQuoteToLocalDatabase: Could not add quote to quoteSlice of localDatabase")
+	}
+
+	// Iterrate over all words of quote
+	for word, count := range getWordsFromString(q.Text) {
+		wordsMapItem := localDatabase.wordsMap[word]
+		wordsMapItem.totalOccurences += count
+
+		wordsMapItem.occurenceSlice = append(wordsMapItem.occurenceSlice, occurenceSliceT{enumid, count})
+
+		localDatabase.wordsMap[word] = wordsMapItem
+	}
+
+	return nil
+}
+
+func unsafeAddQuoteToLocalDatabase(q QuoteT) error {
+
+	localDatabase.quoteSlice = append(localDatabase.quoteSlice, q)
+	var enumid int32 = int32(len(localDatabase.quoteSlice) - 1)
+
+	if enumid < 0 {
+		return errors.New("At addQuoteToLocalDatabase: Could not add quote to quoteSlice of localDatabase")
+	}
+
+	// Iterrate over all words of quote
+	for word, count := range getWordsFromString(q.Text) {
+		wordsMapItem := localDatabase.wordsMap[word]
+		wordsMapItem.totalOccurences += count
+
+		wordsMapItem.occurenceSlice = append(wordsMapItem.occurenceSlice, occurenceSliceT{enumid, count})
+
+		localDatabase.wordsMap[word] = wordsMapItem
+	}
+
+	return nil
+}
+
+func addTeacherToLocalDatabase(t TeacherT) {
+	localDatabase.mux.LockWrite()
+	defer localDatabase.mux.UnlockWrite()
+	localDatabase.teacherSlice = append(localDatabase.teacherSlice, t)
+}
+
+func unsafeAddTeacherToLocalDatabase(t TeacherT) {
+	localDatabase.teacherSlice = append(localDatabase.teacherSlice, t)
+}
+
+func overwriteTeacherInLocalDatabase(t TeacherT) error {
+	localDatabase.mux.LockWrite()
+	defer localDatabase.mux.UnlockWrite()
+
+	affected := false
+	for i, v := range localDatabase.teacherSlice {
+		if v.TeacherID == t.TeacherID {
+			localDatabase.teacherSlice[i] = t
+			affected = true
+			break
+		}
+	}
+
+	if affected == false {
+		return errors.New("At overwriteTeacherInLocalDatabase: Could not find specified entry for overwrite")
+	}
+	return nil
+}
+
+func overwriteQuoteInLocalDatabase(q QuoteT) error {
+	localDatabase.mux.LockWrite()
+	defer localDatabase.mux.UnlockWrite()
+
+	var enumid int32 = -1
+	for i, v := range localDatabase.quoteSlice {
+		if v.QuoteID == q.QuoteID {
+			localDatabase.quoteSlice[i] = q
+			enumid = int32(i)
+		}
+	}
+
+	if enumid < 0 {
+		return errors.New("At overwriteTeacherInLocalDatabase: Could not find specified entry for overwrite")
 	}
 
-	return quotes, nil
+	wordsFromString := getWordsFromString(q.Text)
```

Comment:
what's the purpose of the function from here on? is it to update the wordsMap relating to how the quote changed?
Could this be moved to a different function/file?